### PR TITLE
feat(client): reworked weapon types

### PR DESCRIPTION
**Pull request by HunterNN**

### CODE REVIEW

File: client/logic/game/game.gd [hunk 1]
```diff
@@ -737,8 +737,7 @@ func _do_state_attacking_clicked_left(local: bool = true) -> void:
 		var damage_result: DamageResult
 		damage_result = _calculate_damage(attacking_unit, defending_unit)
 		defending_unit.last_damage_type = damage_result.weapon_type
-		attacking_unit.last_attack_category = damage_result.weapon_category
-		attacking_unit.play_attack()
+		attacking_unit.play_attack(damage_result.weapon_category)
```
Comment (Dracks) [inline client/logic/game/game.gd:740]: I like this!

File: client/logic/game/game.gd [hunk 2]
```diff
@@ -1241,13 +1239,21 @@ func _create_and_set_attack_area(
 				return
 	var terrains: Array[Terrain] = unit.get_possible_terrains_to_attack_from_terrain(target_terrain)
 	for i: Terrain in terrains:
```
Comment (Dracks) [inline client/logic/game/game.gd:1241]: I'm the big fan of one letter variables, but what about having some one letters more descriptives? or maybe better a more long variable? Now the loop grown, I think will be better

File: types/enum.json [hunk 3]
```diff
@@ -134,14 +134,14 @@
...
       "enum": [
-        "ARTILLERY_CANON",
-        "BAZOOKA",
-        "LIGHT_TANK_CANON",
-        "MACHINE_GUN",
-        "MEDIUM_TANK_CANON",
-        "ROCKETS",
-        "TANK_MACHINE_GUN",
-        "VUCLAN_CANNON"
+      "MACHINE_GUN",
```
Comment (Dracks) [inline types/enum.json:137]: the identation changed here, If we start having big json files, maybe we should try to add some json formatter? to keep the identation and everything standard? (only an idea)

**Review by Dracks — APPROVED** [review]

File: client/logic/game/game.gd (hunk 1 above)
Comment (HunterNN) [reply client/logic/game/game.gd:740]: Was thinking the same but the function which uses last_damage_type will be called by an animation (the attacking unit will emit a signal while attacking for every hit which calls that function on the defending unit to show the correct hit effect)

File: types/enum.json (hunk 3 above)
Comment (HunterNN) [reply types/enum.json:137]: Yeah i think so too

**Pull request merged**